File: app/src/main/java/com/example/multidagger/MultiDaggerApplication.java
package com.example.multidagger;

import android.app.Application;
import com.example.multidagger.di.AppInjector;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasAndroidInjector;

import javax.inject.Inject;

public class MultiDaggerApplication extends Application implements HasAndroidInjector {
    @Inject
    volatile DispatchingAndroidInjector<Object> dispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        AppInjector.init(this);
    }

    @Override
    public AndroidInjector<Object> androidInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/kotlin/com/example/multidagger/MultiDaggerApplication.kt
package com.example.multidagger

import com.example.multidagger.di.DaggerAppComponent
import com.example.multidagger.di.applyAutoInjector
import dagger.android.DaggerApplication

class MultiDaggerApplication : DaggerApplication() {

    override fun applicationInjector() = DaggerAppComponent.factory()
            .create(this)

    override fun onCreate() {
        super.onCreate()
        applyAutoInjector()
    }
}
